utils/manifest-utils/src/keystore.rs

```diff
@@ -0,0 +1,306 @@
+    pub added: Vec<Cert>,
+    pub skipped: Vec<Cert>,
+}
+
+pub trait CommonRemoveParams {
+    fn id(&self) -> &HashSet<String>;
+}
+
+pub struct RemoveParams {
+    pub ids: HashSet<String>,
+}
+
+impl CommonRemoveParams for RemoveParams {
+    fn id(&self) -> &HashSet<String> {
+        &self.ids
+    }
+}
+
+#[derive(Default)]
+pub struct RemoveResponse {
+    pub removed: Vec<Cert>,
+}
+
+pub trait Keystore: KeystoreClone + Send {
+    fn reload(&self, cert_dir: &Path) -> anyhow::Result<()>;
+    fn add(&mut self, add: &AddParams) -> anyhow::Result<AddResponse>;
+    fn remove(&mut self, remove: &RemoveParams) -> anyhow::Result<RemoveResponse>;
+    fn list(&self) -> Vec<Cert>;
+}
+
+trait KeystoreBuilder<K: Keystore> {
+    // file is certificate file, certificate is its content
+    fn try_with(&mut self, file: &Path) -> anyhow::Result<()>;
+    fn build(self) -> anyhow::Result<K>;
+}
+
+pub trait KeystoreClone {
+    fn clone_box(&self) -> Box<dyn Keystore>;
+}
+
+impl<T> KeystoreClone for T
+where
+    T: 'static + Keystore + Clone,
+{
+    fn clone_box(&self) -> Box<dyn Keystore> {
+        Box::new(self.clone())
+    }
+}
+
+impl Clone for Box<dyn Keystore> {
+    fn clone(&self) -> Box<dyn Keystore> {
+        self.clone_box()
+    }
+}
+
+#[derive(Clone)]
+pub struct CompositeKeystore {
+    x509_keystore: X509KeystoreManager,
+    golem_keystore: GolemKeystore,
+}
+
+impl CompositeKeystore {
+    pub fn load(cert_dir: &PathBuf) -> anyhow::Result<Self> {
+        std::fs::create_dir_all(cert_dir)?;
+        let mut x509_builder = X509KeystoreBuilder::new(cert_dir);
+        let mut golem_builder = GolemKeystoreBuilder::new(cert_dir);
+
+        let cert_dir = std::fs::read_dir(cert_dir)?;
+        for dir_entry in cert_dir {
+            let file = dir_entry?;
+            let file = file.path();
+            if let Err(err) = golem_builder
+                .try_with(&file)
+                .or_else(|err| {
+                    log::trace!("File '{} is not a Golem certificate. Trying to parse as Golem certificate. Err: {err}'", file.display());
+                    x509_builder.try_with(&file)
+                }) {
+                    log::debug!("File '{} is not a X509 certificate nor a Golem certificate. Err: {err}'", file.display());
+                }
+        }
+
+        let x509_keystore = x509_builder.build()?;
+        let golem_keystore = golem_builder.build()?;
+
+        Ok(Self {
+            x509_keystore,
+            golem_keystore,
+        })
+    }
+
+    fn keystores(&self) -> Vec<&dyn Keystore> {
+        vec![&self.golem_keystore, &self.x509_keystore]
+    }
+
+    fn keystores_mut(&mut self) -> Vec<&mut dyn Keystore> {
+        vec![&mut self.golem_keystore, &mut self.x509_keystore]
+    }
+
+    pub fn list_ids(&self) -> HashSet<String> {
+        self.list()
+            .into_iter()
+            .map(|cert| cert.id())
+            .collect::<HashSet<String>>()
+    }
+
+    pub fn verify_signature(
```

Comment:
```suggestion
    pub fn verify_x509_signature(
```

I propose this change to expicit say what we are verifying